### PR TITLE
Unit test infrastructure and first test

### PR DESCRIPTION
**Pull request by eslavich**
Here's some test stuff.  This PR adds some fixtures for creating Box Files and Folders, tox config with black, flake8, pytest, and pytest-cov checks, and a single test.  It also reformats the redirector code a bit to keep black happy.

### CODE REVIEW

File: setup.cfg [hunk 1]
```diff
@@ -0,0 +1,36 @@
+[metadata]
+name = redirector
+provides = redirector
+package_name = redirector
+author = STScI
+author_email = help@stsci.edu
+license = BSD 3-Clause
+license_file = LICENSE
+description = A package for syncing JIRA tickets and GitHub issues
```
Comment (bhayden53) [inline setup.cfg]: copy/paste?

File: setup.cfg [hunk 2]
```diff
@@ -0,0 +1,36 @@
+[metadata]
+name = redirector
+provides = redirector
+package_name = redirector
+author = STScI
+author_email = help@stsci.edu
```
Comment (bhayden53) [inline setup.cfg]: Is this right?

File: tests/conftest.py [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+    return _create_folder
+
+
+def _next_box_object_id():
+    result = str(_next_box_object_id._next_id)
+    _next_box_object_id._next_id += 1
+    return result
+
+
+_next_box_object_id._next_id = 500000000000
```
Comment (bhayden53) [inline tests/conftest.py]: Can you comment this?

File: setup.cfg (hunk 1 above)
Comment (eslavich) [reply setup.cfg]: Oops, thanks

File: setup.cfg (hunk 2 above)
Comment (eslavich) [reply setup.cfg]: I'm not sure, but I see it in other repos.  Happy to change it if you can think of a better option.

File: tests/conftest.py (hunk 3 above)
Comment (eslavich) [reply tests/conftest.py]: Yup!

**Pull request merged**